tests/fsharpqa/Source/Conformance/InferenceProcedures/NameResolution/RequireQualifiedAccess/OnRecordVsUnion.fs

```diff
@@ -0,0 +1,11 @@
+// #Conformance #TypeInference #Attributes 
+// Verify the RequireQualifiedAccess attribute works on unions
+
+module A =
+    [<RequireQualifiedAccess>]
+    type U = | C
+
+    type C() =
+        static member M() = ()
+
+let x = A.C.M()
```

Comment:
I'm trying to understand what this test adds.  Today it gives a warning and then doesn't compile.  What does it give after this PR?  Does it just compile?

Likewise, what happens after this PR without the RQA attribute (and with the last line adjusted to ``let x : A.C = A.C``)?  Today that compiles without warning?  But it feels to me that after this change then it won't compile any more?  I haven't tried yet, just wondering.  

Adding these test cases would be good, and also adding test cases for union types where the type name is the same as the union case name, e.g. ``type C = | C``